src/dev/build/tasks/fetch_agent_versions_list.ts

```diff
@@ -33,8 +39,13 @@ const getAvailableVersions = async (log: ToolingLog) => {
     log.info(`Retrieved available versions`);
     return versions;
   } catch (error) {
-    log.warning(`Failed to fetch versions list`);
-    log.warning(error);
+    log.warning(`Failed to fetch Elastic Agent versions list`);
+
+    if (process.env.BUILDKITE_PULL_REQUEST === 'true') {
+      log.warning(error);
+    } else {
+      throw new Error(error);
```

Comment:
Does the error will still be logged in that case? if it's not the case maybe we want to log the error before throwing